Update immutability blueprint documentation (#4)
lucas-tortora/blueprints: doc-immutability/overview.md

# Document Immutability

**Documents are an important means of transporting information and contracts. Being able to reliably prove that a document has not been tampered from an established state helps ensure trust between the parties envolved. This blueprint describes a solution that enables users to verify data integrity of a document using the IOTA Tangle, while remaining in complete control of the data.**
## Business Case

### Challenges

> Ensuring the integrity of digital documents is currently mostly done by storing them in some trusted database or by physically printing the documents. However, in a trustless and efficient environment, this is far from optimal.

Reviewer: ```suggestion
Most people ensure the integrity of digital documents by storing them in some trusted database or physically printing them. However, in a trustless and efficient environment, this is far from optimal.
```

> A step in the right direction is e.g. displaying data fingerprints, so called hashes, on a website from which a file is being downloaded, which can then manually be compared to the local copy's fingerprint by a technically versed user. 

Reviewer: ```suggestion
Displaying data fingerprints (hashes) on a website that hosts the document which the user can compare to the local copy's fingerprint is a step in the right direction.
```

> However, this procedure is too technically challenging and time consuming for most users. Additionaly, displaying a hash on some website is far from fully secure, as the centralized database of that website is also subject to change. For example, in 2016, hackers [successfully put a backdoor](https://blog.linuxmint.com/?p=2994) into an ISO file of a Linux distribution called Mint. As they also managed to get access to the website, they were able to adjust the hashes being displayed on the website too, thereby completely bypassing this security mechanism.

Reviewer: ```suggestion
However, this process can be technically challenging and time-consuming for most users. Additionally, as the centralized database of that website is also subject to change, displaying a hash on is far from fully secure. For example, in 2016, [hackers successfully put a backdoor](https://blog.linuxmint.com/?p=2994) into an ISO file of a Linux distribution called Mint. As they also managed to get access to the website, they were able to adjust the hashes displayed there, completely bypassing this security mechanism.
```

> Also, in a scenario not containing some centralized server, but two parties agreeing on a contract, a simple tool should exist that anchors the integrity of a document in a distributed storage. 

Reviewer: ```suggestion
In a scenario that involves two parties agreeing on a contract instead of a centralized server, a simple tool should exist that anchors the integrity of a document in distributed storage. 
```

### Solution

> In this blueprint, we leverage the distributed ledger of the IOTA Tangle to immutably store data fingerprints of files, thereby providing a way to verify the integrity of a document by not relying on a centralized instance. 

Reviewer: ```suggestion
This blueprint leverages the IOTA Tangle as a distributed ledger to immutably store data fingerprints of files, providing a way to verify the integrity of a document that does not rely on a centralized instance. 
```

This enables users to prove that a document has not been tampered with since the data fingerprint ("Proof of Existence") has been stored on the Tangle.

> To verify a document in our web application, users can provide a file and a message-id, used to locate the data fingerprint embedded in a message of the Tangle. The provided file is locally hashed and compared with the fetched fingerprint stored on the Tangle.

Reviewer: ```suggestion
To verify a document in the web application, users can provide a file and a message-id.  The application will then use the message-id to locate the data fingerprint embedded in a message on the Tangle. The provided file is locally hashed and compared with the fingerprint which is stored on the Tangle.
```

Users can also publish new data fingerprints of a file and receive a message-id that references the Proof of Existence in the Tangle.

> Additionally, the users remain in complete control of the document, meaning that the document itself is never exposed to the the public. The hash that is stored on the Tangle can be used to verify the integrity of the document, however the underlying document can not be derived from it. 

Reviewer: ```suggestion
Additionally, as the Tangle never exposes the document to the public, the users remain in complete control of the document. An application can use the hash stored on the Tangle to verify the document's integrity, but the underlying document can not be derived from it. 
```

## Demo

> See this website for a [demonstration of a Proof of Existence application](https://iota-poex.dag.sh/).

Reviewer: ```suggestion
You can find a live demo of a Proof of Existence application in the [Immutable Document Blueprint website](https://iota-poex.dag.sh/).
```
I'm not quite sure about the site's title.

Author: It is not really a website, so I think we should stick with the first one :)

> Check the [GitHub Repository](https://github.com/iotaledger/poc-document-immutable-blueprint) for instructions on how to set up a local version of this blueprint.

Reviewer: ```suggestion
You can find instructions on how to set up a local version of this blueprint in the [official GitHub repository](https://github.com/iotaledger/poc-document-immutable-blueprint).
```

## Additional Resources

---------------
### [GitHub Repository](https://github.com/iotaledger/poc-document-immutable-blueprint)

> Access the code base and read some quickstart instructions to test this blueprint.

Reviewer: ```suggestion
Access the code base and quickstart instructions to test this blueprint.
```

---

### [Proof of Existence Library](https://www.npmjs.com/package/@iota/poex-tool)

> A library that can be used for Proof of Existence scenarios that is also used in this blueprint.

Reviewer: ```suggestion
A library you can use for Proof of Existence scenarios that is also used in this blueprint.
```
